src/main/java/com/github/manolo8/darkbot/gui/tree/editors/NewShipModeEditor.java

```diff
@@ -0,0 +1,96 @@
+package com.github.manolo8.darkbot.gui.tree.editors;
+
+import com.github.manolo8.darkbot.gui.AdvancedConfig;
+import com.github.manolo8.darkbot.gui.tree.utils.SizedLabel;
+
+import eu.darkbot.api.config.ConfigSetting;
+import eu.darkbot.api.config.types.ShipMode;
+import eu.darkbot.api.config.util.OptionEditor;
+import eu.darkbot.api.game.items.SelectableItem.Formation;
+import eu.darkbot.api.managers.HeroAPI;
+
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.event.ItemEvent;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class NewShipModeEditor extends JPanel implements OptionEditor<ShipMode> {
```

Comment:
ideally this should be merged into the priorly existing ship config editor, there's no need for them to be separate